### PR TITLE
feat: add pool like

### PR DESCRIPTION
**Pull request by 2fd**

### CODE REVIEW

File: src/PoolLike/PoolLike.types.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export type PoolLikeAttributes = {
+  pool: string
+  user: string
+  created_at: string
+}
+
+export const poolSchema = {
```
Comment (nicosantangelo) [inline src/PoolLike/PoolLike.types.ts]: I think this should be `poolLikeSchema`

File: src/PoolLike/PoolLike.router.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    const exists = await PoolLike.count({ pool, user })
+
+    if (!exists) {
+      await PoolLike.create({ pool, user, created_at: new Date() })
+      const likes = await PoolLike.count({ pool })
+      await Pool.update({ likes }, { id: pool })
+    }
+
+    return true
```
Comment (nicosantangelo) [inline src/PoolLike/PoolLike.router.ts]: Wouldn't it be useful/necessary to return a more comprehensive response? for example returning the PoolLike record, or maybe the total count?

That way the consumer could see if the like already existed for example

File: src/PoolLike/PoolLike.router.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    const exists = await PoolLike.count({ pool, user })
+
+    if (exists) {
+      await PoolLike.delete({ pool, user })
+      const likes = await PoolLike.count({ pool })
+      await Pool.update({ likes }, { id: pool })
+    }
+
+    return true
```
Comment (nicosantangelo) [inline src/PoolLike/PoolLike.router.ts]: Same as above! maybe it's useful to return a more comprehensive response?

File: src/Pool/Pool.router.ts [hunk 4]
```diff
@@ -78,11 +92,23 @@ export class PoolRouter extends Router {
   }
 
   async getPool(req: AuthRequest) {
-    const id = server.extractFromReq(req, 'id')
-    return Pool.findOne({ id })
+    const pool_id = server.extractFromReq(req, 'id')
+    const user_id = (req.auth && req.auth.sub) || null
```
Comment (nicosantangelo) [inline src/Pool/Pool.router.ts]: I think we're doing this: `requestParameters.get('user_id', null)` on `getPools`. Maybe we can do the same here?

Also, as a TODO (not now), we might want to have a PermissiveAuthRequest or something of the sort for these casese

File: src/PoolLike/PoolLike.types.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  user_id: string
+  created_at: string
+}
+
+export type PoolLikeCount = {
+  pool_id: string
+  user_id?: string
+}
+
+export const poolLikeSchema = {
```
Comment (nicosantangelo) [inline src/PoolLike/PoolLike.types.ts]: Should we delete `poolLikeSchema`? it's not really being used. Or should we use it in the corresponding endpoint?

File: src/Pool/Pool.router.ts (hunk 4 above)
Comment (2fd) [reply src/Pool/Pool.router.ts]: This expression is for determining if the current user (if there is) gives like to the pool. You don't want to load a pool with the like of an arbitrary user

**Pull request merged**